controllers/article-controllers.js

```diff
@@ -17,4 +27,22 @@ const getArticle = (request, response, next) => {
     });
 };
 
-module.exports = { getArticle, getAllArticles };
+const getArticleComments = (request, response, next) => {
+  const { article_id } = request.params;
+  const promises = [fetchArticleComments(article_id)];
+
+  if (article_id) {
+    promises.push(checkIdExists(article_id));
+  }
+
+  Promise.all(promises)
+    .then((resolvedPromises) => {
+      const comments = resolvedPromises[0];
+      response.status(200).send({ comments });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
```

Comment:
You should try and handle your data in the model rather than the controller, so that the data is ready for your response when it leaves the model. You could deconstruct from rows in the model too and then return the comments array with only one level of nesting.